src/components/VmConsole/index.js

```diff
@@ -0,0 +1,128 @@
+import disconnectIcon from './images/disconnect_icon.svg'
+
+const RDP_ID = 'rdp'
+
+const InfoPageContainer = ({ mainText, secondaryText, icon, secondaryComponent }) => (
+  <div className={style['download-page-container']}>
+    <div className={style['download-page-content']}>
+      <img src={icon} />
+      <span className={style['download-page-main-text']}>{ mainText }</span>
+      <span className={style['download-page-secondary-text']}>{ secondaryText }</span>
+      { secondaryComponent &&
+        <span className={style['download-page-secondary-text']}>
+          { secondaryComponent }
+        </span>
+      }
+    </div>
+  </div>
+)
+
+InfoPageContainer.propTypes = {
+  mainText: PropTypes.string.isRequired,
+  secondaryText: PropTypes.string.isRequired,
+  icon: PropTypes.string.isRequired,
+  secondaryComponent: PropTypes.node,
+}
+
+class VmConsole extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      vmId: props.vmId,
+      consoleId: props.consoleId,
+      isFirstRun: props.consoles.getIn(['vms', props.vmId]) === undefined,
+    }
+  }
+
+  componentDidMount () {
+    if (this.state.isFirstRun && this.props.consoleId === RDP_ID) {
+      const domain = this.props.config.get('domain')
+      const username = this.props.config.getIn([ 'user', 'name' ])
+      this.props.onRDP({ domain, username, vms: this.props.vms })
+      this.setState({ isFirstRun: false })
+    }
+  }
+
+  render () {
+    const { vmId, config, consoleId, vms, onDisconnected } = this.props
+    const websocket = config.get('websocket')
+    if (consoleId === RDP_ID) {
+      return <InfoPageContainer icon={downloadIcon} mainText={msg.downloadedRDPFile()} secondaryText={msg.downloadedRDP()} />
+    }
+    const currentConsole = vms.getIn(['vms', vmId, 'consoles']).find(c => c.get('id') === consoleId)
+    if (this.state.isFirstRun) {
+      return <div>
+        <Loader loaderText='Loading' size={SIZES.SMALL} />
+        <ConsoleConfirmationModal isConsolePage isNoVNC vm={vms.getIn(['vms', vmId])} consoleId={consoleId} onClose={() => this.setState({ isFirstRun: false })} show />
+      </div>
+    }
+    const proxyTicket = this.props.consoles.getIn(['vms', vmId, 'proxyTicket'])
+    const ticket = this.props.consoles.getIn(['vms', vmId, 'ticket'])
+    switch (this.props.consoles.getIn(['vms', vmId, 'consoleStatus'])) {
+      case INIT_CONSOLE:
+        if (ticket !== undefined && websocket !== null) {
+          return <VncConsole
+            encrypt
+            resizeSession
+            scaleViewport
+            textConnecting={<Loader loaderText='Connecting' size={SIZES.SMALL} />}
+            credentials={{ password: ticket.value }}
+            path={proxyTicket}
+            host={websocket.get('host')}
+            port={websocket.get('port')}
+            portalToolbarTo='vm-console-toolbar-sendkeys'
+            onDisconnected={onDisconnected}
+          />
```

Comment:
- `textConnecting` -- the `loaderText` needs localization
  - `textDisconnected` :heavy_check_mark: -- ah ha... it isn't needed since rendering the disconnected state is handled in `VmConsole`
  - `textSendShortcut` :heavy_check_mark: 
  - `textCtrlAltDel` :heavy_check_mark: 
  - `textDisconnect` :heavy_check_mark: